grails-app/controllers/org/pih/warehouse/JsonController.groovy

```diff
@@ -1777,48 +1777,50 @@ class JsonController {
     }
 
     def getRequestDetailReport = {
-        DateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy")
-        params.startDate = dateFormat.parse(params.startDate)
-        params.endDate = dateFormat.parse(params.endDate)
-        def data = forecastingService.getRequestDetailReport(params)
-        if (params.format == "text/csv") {
-            def sw = new StringWriter()
-
-            def csv = new CSVWriter(sw, {
-                "Request Number" { it.requestNumber }
-                "Date Requested" { it.dateRequested }
-                "Date Issued" { it.dateIssued }
-                "Origin" { it.origin }
-                "Destination" { it.destination }
-                "Product Code" { it.productCode }
-                "Product Name" { it.productName }
-                "Quantity Requested" { it.qtyRequested }
-                "Quantity Issued" { it.qtyIssued }
-                "Reason Code" { it.reasonCode }
-                "Quantity Demand" { it.qtyDemand }
-            })
-
-            data.each {
-                csv << [
-                        requestNumber  : it.requestNumber,
-                        dateRequested  : it.dateRequested,
-                        dateIssued  : it.dateIssued,
-                        origin    : StringEscapeUtils.escapeCsv(it.origin),
-                        destination: StringEscapeUtils.escapeCsv(it.destination),
-                        productCode  : it.productCode,
-                        productName  : StringEscapeUtils.escapeCsv(it.productName),
-                        qtyRequested     : it.quantityRequested,
-                        qtyIssued     : it.quantityIssued,
-                        reasonCode     : it.reasonCode,
-                        qtyDemand     : it.quantityDemand,
-                ]
-            }
+        if (params.startDate && params.endDate) {
```

Comment:
What will happen if there will be no startDate and endDate? You should throw some error in that case (I know that there is validation on the front end, but someone can use this method in another place without proper validation).